c_binding/include/libstoragemgmt/libstoragemgmt.h

```diff
@@ -37,111 +37,217 @@
- * @param[in] flags     Reserved for future use, must be zero.
- * @return Error code as enumerated by \ref lsm_error_number.
- * @retval LSM_ERR_OK on success.
+ * lsm_connect_password - Get a connection to a storage provider.
+ *
+ * Version:
+ *      1.0
+ *
+ * Description:
+ *      Get a connection to a storage provider.
+ *
+ * @uri:
+ *      Uniform Resource Identifier (see URI documentation)
+ * @password:
+ *      Password for the storage array (optional, can be NULL)
+ * @conn:
+ *      The connection to use for all the other library calls
+ * @timeout:
+ *      Time-out in milliseconds, (initial value).
+ * @e:
+ *      Error data if connection failed.
+ * @flags:
+ *      Reserved for future use, must be LSM_CLIENT_FLAG_RSVD.
+ *
+ * Return:
+ *      Error code as enumerated by 'lsm_error_number'.
+ *          * LSM_ERR_OK
+ *              On success.
+ *          * LSM_ERR_INVALID_ARGUMENT
+ *              When any argument is NULL or invalid flags.
  */
 int LSM_DLL_EXPORT lsm_connect_password(const char *uri,
                                         const char *password,
                                         lsm_connect **conn,
                                         uint32_t timeout,
                                         lsm_error_ptr *e, lsm_flag flags);
 /**
- * Closes a connection to a storage provider.
- * @param[in] conn      Valid connection to close
- * @param[in] flags     Reserved for future use, must be zero.
- * @return Error code as enumerated by \ref lsm_error_number.
- * @retval LSM_ERR_OK on success.
+ * lsm_connect_close - Closes a connection to a storage provider.
+ *
+ * Version:
+ *      1.0
+ *
+ * Description:
+ *      Clones a connection to a storage provisioning.
+ *
+ * @conn:
+ *      Valid connection to close
+ * @flags:
+ *      Reserved for future use, must be LSM_CLIENT_FLAG_RSVD.
+ *
+ * Return:
+ *      Error code as enumerated by 'lsm_error_number'.
+ *          * LSM_ERR_OK
+ *              On success.
+ *          * LSM_ERR_INVALID_ARGUMENT
+ *              When not a valid lsm_connect pointer or invalid flags.
  */
 int LSM_DLL_EXPORT lsm_connect_close(lsm_connect *conn, lsm_flag flags);
 
 /**
- * Retrieve information about the plug-in
- * NOTE: Caller needs to free desc and version!
- * @param[in] conn      Valid connection @see lsm_connect_password
- * @param[out] desc     Plug-in description
- * @param[out] version  Plug-in version
- * @param [in] flags    Reserved for future use, must be zero.
- * @return Error code as enumerated by \ref lsm_error_number.
- * @retval LSM_ERR_OK on success.
+ * lsm_plugin_info_get - Retrieves information about the plug-in
+ *
+ * Version:
+ *      1.0
+ *
+ * Description:
+ *      Retrieves information about the plug-in.
+ *
+ * @conn:
+ *      Valid lsm_connect pointer.
+ * @desc:
+ *      Plug-in description. Returned string should be freed manually.
+ * @version:
+ *      Plug-in version. Returned string should be freed manually.
```

Comment:
Plug-in version. Returned string should be freed manually with `free` library call.